yt_dlp/extractor/acfun.py

```diff
@@ -0,0 +1,192 @@
+
+        codec_str = info.get('codecs') or ''
+        m = re.match('(?P<vc>[^,]+),(?P<ac>[^,]+)', codec_str)
+        if m:
+            vcodec = m.group("vc")
+            acodec = m.group("ac")
+
+        formats, _ = self._extract_m3u8_formats_and_subtitles(
+            info['url'], video_id)
+
+        # it seems AcFun do not have subtitles
+
+        return {
+            **formats[0],
+            'fps': int_or_none(info.get('frameRate')),
+            'width': int_or_none(info.get('width')),
+            'height': int_or_none(info.get('height')),
+            'vcodec': vcodec,
+            'acodec': acodec,
+            'tbr': float_or_none(info.get('avgBitrate')),
+            'ext': 'mp4'
+        }
+
+    def parse_format_list(self, jobj, video_id):
+        formats = []
+        for video in jobj:
+            fmt = self.parse_format(video, video_id)
+            formats.append(fmt)
+
+        self._sort_formats(formats)
+
+        return formats
+
+    def gen_other_video_info_map(self, video_info):
+        return {
+            'duration': float_or_none(video_info.get('durationMillis'), 1000),
+            'timestamp': int_or_none(video_info.get('uploadTime'), 1000),
+        }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+        json_all = self._search_json(r'window.videoInfo\s*=\s*', webpage, 'videoInfo', video_id)
+
+        video_info = json_all['currentVideoInfo']
+        playjson = self._parse_json(video_info['ksPlayJson'], video_id)
+        video_internal_id = traverse_obj(json_all, ('currentVideoInfo', 'id'))
+
+        formats = self.parse_format_list(traverse_obj(playjson, ('adaptationSet', 0, 'representation')), video_id)
+
+        video_list = json_all['videoList']
+        p_idx, video_info = [(idx + 1, v) for (idx, v) in enumerate(video_list)
+                             if v['id'] == video_internal_id
+                             ][0]
+
+        title = json_all['title']
+        if len(video_list) > 1:
+            title = f"{title} P{p_idx:02d} {video_info['title']}"
+
+        return {
+            'id': video_id,
+            'title': title,
+            'formats': formats,
+            'thumbnail': json_all.get('coverUrl'),
+            'description': json_all.get('description'),
+            'uploader': traverse_obj(json_all, ('user', 'name')),
+            'uploader_id': traverse_obj(json_all, ('user', 'href')),
+            'tags': [t['name'] for t in json_all.get('tagList', []) if 'name' in t],
+            'view_count': int_or_none(json_all.get('viewCount')),
+            'like_count': int_or_none(json_all.get('likeCountShow')),
+            'comment_count': int_or_none(json_all.get('commentCountShow')),
+            'http_headers': {
+                'Referer': url,
+            },
+            **self.gen_other_video_info_map(video_info)
+        }
+
+
+class AcFunBangumiIE(AcFunVideoIE):
```

Comment:
Do not inherit from a concrete IE. Instead, create a BaseIE from which both inherits